docs/multiarch.rst

```diff
@@ -0,0 +1,1105 @@
+repository should be changed so that it no longer specifies any
+architecture, as atomic-reactor will supply this field.
+
+Having the orchestrator build do this step, which mostly involves
+waiting for the Koji task to finish, results in better (more accurate)
+resource allocation. Orchestrator builds will have slimmer resource
+requests than those of worker builds.
+
+After the Koji task has finished, the worker builds then need to be
+instructed to take their input from a specific output of that
+task. The ``add_filesystem`` plugin will need changes for this:
+
+- it will need a parameter to tell it to create a multi-platform
+  image-build task and not stream the output of that task. This
+  parameter will be set for the orchestrator build.
+
+- it can already be told to take its input from the output of a
+  specific Koji task, but will need to be able to decide which
+  particular task output file is required by parsing the output
+  filenames and looking for the platform name. This parameter will be
+  set for the worker build.
+
+Excluding platforms
+-------------------
+
+Some container images will need to be built for multiple platforms but
+some may not.
+
+The ``osbs.conf`` file will specify the full list of platforms for which
+worker builds should be created. This list of platforms can be reduced
+in three ways:
+
+- Including platforms:
+
+  * the set of platforms an image is built for is determined by the
+    Koji build tag for the Koji build target
+
+  * Container image builds can be submitting with a parameter
+    ``--arches``, overriding the set of platforms specified by the Koji
+    build target, in the same way as for building RPM packages
+
+- Excluding platforms:
+
+  * a git repository file ``exclude-platform`` can be used to list
+    platforms which this image should not be built for
+
+Tagging
+-------
+
+There are no changes to the tagging scheme. Each image manifest will
+be tagged as before using:
+
+- ``latest``
+- $version
+- $version-$release
+- a unique tag including the timestamp
+- any additional tags configured in the git repository
+
+Scratch builds
+--------------
+
+There are no changes to how scratch builds are performed, only some
+parts of the implementation will move around. Some build steps will be
+omitted when performing scratch builds:
+
+- only unique tags will be applied in the worker builds
+- the result will not be imported into Koji in the orchestrator build
+
+Streamed build logs
+-------------------
+
+When atomic-reactor in the orchestrator build runs its
+``orchestrate_build`` plugin and watches the builds, it will stream logs
+from those builds and log them again itself with a special prefix
+indicating they are logs relating to a particular platform.
+
+Note that there will be a single Koji task with multiple log
+outputs. When watching this using ``koji watch-logs <task id>`` the log
+output from each worker build will be interleaved. To watch logs from
```

Comment:
This might not be very useful most of the time. How is this currently working in koji?

A more useful approach would be that in cases when multiple builds are running for the same task, `koji watch-logs <task id> should only list the available log file names, so that the user can easily pick one of them to watch.